src/utils/common.js

```diff
@@ -222,16 +222,32 @@ export function download(downloadUrl, filename) {
   const iframe = document.createElement('iframe')
   iframe.style.display = 'none'
   document.body.appendChild(iframe)
-  const a = document.createElement('a')
-  a.href = downloadUrl
+  const timeout = 1000 * 60 * 30
+
   if (filename) {
-    a.download = filename
+    fetch(downloadUrl)
+      .then(response => response.blob())
+      .then(blob => {
+        const url = URL.createObjectURL(blob)
+        const a = iframe.contentWindow.document.createElement('a')
+        a.href = url
+        a.download = filename
+        iframe.contentWindow.document.body.appendChild(a)
+        a.click()
+        setTimeout(() => {
+          URL.revokeObjectURL(url)
+          document.body.removeChild(iframe)
+        }, timeout) // If you can't download it in half an hour, don't download it.
+      })
+      .catch(() => {
+        document.body.removeChild(iframe)
+      })
+  } else {
+    iframe.src = downloadUrl
+    setTimeout(() => {
+      document.body.removeChild(iframe)
+    }, timeout) // If you can't download it in half an hour, don't download it.
   }
-  iframe.contentWindow.document.body.appendChild(a)
-  a.click()
-  setTimeout(() => {
-    document.body.removeChild(iframe)
-  }, 1000 * 60 * 30) // If you can't download it in half an hour, don't download it.
 }
 
 export function diffObject(object, base) {
```

Comment:
The given code seems to be a snippet of JavaScript that uses `iframe` elements to open web pages containing links to files for downloading. It appears that there aren't any explicit syntax errors or logical issues in the script provided.

However, here are some general recommendations and points of consideration:

1. **Compatibility Check**: Ensure that `iframe`, both for browser compatiblity reasons including Firefox and Opera's support, use version 0 or later (`<iframe></iframe>`) to avoid any security risks due to older versions being susceptible; consider checking which one works best with browsers used by users.

2. **Security**: The current implementation uses blob URLs explicitly instead of using file paths directly, though this could lead to performance issues as it may require HTTP/HTTPS handling when reading the file content. Using `blob()` method also makes it vulnerable if server responses cannot meet expectations (like missing `.html`, etc.). A more secure approach might involve passing along data via a JSON object where only the needed fields are included.

3. **Timeout Handling**: When trying to download a file without specifying a specific filename and after attempting three times within a minute, setting a time out does not seem appropriate. Timeout values like 5 minutes should be implemented carefully considering the user experience and site context.

4. **Async Loading**: For future modifications, considering implementing asynchronous loading would make sense as modern websites typically leverage async resources efficiently.

5. **Documentation Improvement**: Include comments and clear documentation on what each part of the functions does, especially how they interact and handle errors.

In summary, while there appear to be no immediate glaring problems, there is room for refinement and additional considerations for better code quality.